src/backend/aspen/workflows/nextstrain_run/export.py

```diff
@@ -127,146 +131,135 @@ def cli(
+                    )
                 ),
-                "genbank_accession": pathogen_genome_id_repository_type_to_accession_names.get(
-                    (pathogen_genome.entity_id, PublicRepositoryType.GENBANK), ""
+                accession_input_alias.id.in_(
+                    {pathogen_genome.entity_id for pathogen_genome in pathogen_genomes}
                 ),
-                "date": sample.collection_date.strftime("%Y-%m-%d"),
-                "date_submitted": upload_date,
-                "region": sample.region.value,
-                "country": sample.country,
-                "division": sample.division,
-                "location": sample.location,
-                "region_exposure": sample.region.value,
-                "country_exposure": sample.country,
-                "division_exposure": sample.division,
-                "segment": "genome",
-                "length": len(sequence),
-                "host": "Human",
-                "age": "?",
-                "sex": "?",
-                "originating_lab": sample.sample_collected_by,
-                "submitting_lab": sample.submitting_group.name,
-                "authors": ", ".join(sample.authors),
-            }
+            )
+        )
+    }
 
-            metadata_csv_fh.writerow(aspen_metadata_row)
-            sequences_fh.write(f">{sample.public_identifier}\n")
-            sequences_fh.write(sequence)
-            sequences_fh.write("\n")
+    aspen_samples: Set[str] = set()
+    metadata_csv_fh = csv.DictWriter(metadata_fh, METADATA_CSV_FIELDS, delimiter="\t")
+    metadata_csv_fh.writeheader()
+    for pathogen_genome in pathogen_genomes:
+        # find the corresponding sample
+        if isinstance(pathogen_genome, UploadedPathogenGenome):
+            sample_id = pathogen_genome.sample_id
+        else:
+            raise ValueError("pathogen genome of unknown type")
+        sample = sample_id_to_sample[sample_id]
+        aspen_samples.add(sample.public_identifier)
 
-        print(
-            json.dumps(
-                {
-                    "bucket": aligned_gisaid.s3_bucket,
-                    "metadata_key": aligned_gisaid.metadata_s3_key,
-                    "sequences_key": aligned_gisaid.sequences_s3_key,
-                }
-            )
+        sequence = "".join(
+            [
+                line
+                for line in pathogen_genome.sequence.splitlines()
+                if not (line.startswith(">") or line.startswith(";"))
+            ]
         )
+        sequence = sequence.strip("Nn")
+
+        upload_date = None
+        if sample.sequencing_reads_collection is not None:
+            upload_date = sample.sequencing_reads_collection.upload_date.strftime(
+                "%Y-%m-%d"
+            )
+        elif sample.uploaded_pathogen_genome is not None:
+            upload_date = sample.uploaded_pathogen_genome.upload_date.strftime(
+                "%Y-%m-%d"
+            )
```

Comment:
why not put upload_date = None in an else block? I think that's a little easier to read

Reply:
@jgadling wrote this part, but I also like it this way because it's a lot more clear that `upload_date` is in the scope of the larger block; and it shows the default value first.

Reply:
I much prefer a default value above a check for a special case!

That said though, I think we're skipping support for sequencing_reads_collections at the moment, so we can assume we'll always have an uploaded pathogen genome?

Reply:
There's the rare occurrence of a legacy failed sample that wouldn't have the uploaded pathogen genome so this is a good check